src/bundle/Resources/config/services/suggestions.yaml

```diff
@@ -0,0 +1,36 @@
+parameters:
+    ibexa.site_access.config.default.search.suggestion.min_query_length: 3
+    ibexa.site_access.config.default.search.suggestion.result_limit: 5
+
+services:
+    _defaults:
+        autoconfigure: true
+        autowire: true
+        public: false
+
+    Ibexa\Bundle\Search\Controller\SuggestionController:
+        tags:
+            - { name: 'container.service_subscriber', key: 'serializer', id: 'ibexa.search.suggestion.serializer' }
+
+    Ibexa\Bundle\Search\ArgumentResolver\SuggestionQueryArgumentResolver:
+        arguments:
+            $defaultLimit: '%ibexa.site_access.config.default.search.suggestion.result_limit%'
+        tags:
+            - { name: 'controller.argument_value_resolver', priority: 50 }
+
+    Ibexa\Search\EventDispatcher\EventListener\ContentSuggestionSubscriber: ~
+
+    Ibexa\Search\Mapper\SearchHitToContentSuggestionMapper: ~
+
+    Ibexa\Contracts\Search\Mapper\SearchHitToContentSuggestionMapperInterface: '@Ibexa\Search\Mapper\SearchHitToContentSuggestionMapper'
+
+    Ibexa\Search\Service\SuggestionService: ~
+
+    Ibexa\Contracts\Search\Service\SuggestionServiceInterface: '@Ibexa\Search\Service\SuggestionService'
+
+    Ibexa\Search\Service\Event\SuggestionService:
+        decorates: Ibexa\Contracts\Search\Service\SuggestionServiceInterface
```

Comment:
Something for the team: we might want to think about setting a priority here, so that events are either fired by default (by higher priority) or overwritten (by lower priority). Without setting this and having priority remain 0, the order of decoration will be undefined and unpredictable.